### PR TITLE
XML parser integration test

### PR DESCRIPTION
**Pull request by ananya-bhattacharya**
Integration test for XMLParser.
Build success on cluster(104.155.156.158) and standalone:
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 4:08.545s
[INFO] Finished at: Fri Jun 24 17:29:37 IST 2016
[INFO] Final Memory: 79M/333M
[INFO] ------------------------------------------------------------------------

### CODE REVIEW

File: cdap-integration-tests/integration-test-remote/src/test/java/co/cask/cdap/app/etl/batch/XMLParserTest.java [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+                    Schema.Field.of("subcategory", Schema.nullableOf(Schema.of(Schema.Type.STRING))));
+
+
+  @Test
+  public void test() throws Exception {
+    ETLStage source = new ETLStage("TableSource", new ETLPlugin("Table", BatchSource.PLUGIN_TYPE, ImmutableMap.of(
+      Properties.BatchReadableWritable.NAME, XMLPARSER_SOURCE,
+      Properties.Table.PROPERTY_SCHEMA, SOURCE_SCHEMA.toString()), null));
+
+    Map<String, String> sourceproperties = new ImmutableMap.Builder<String, String>()
```
Comment (bhushankawadkar) [inline cdap-integration-tests/integration-test-remote/src/test/java/co/cask/cdap/app/etl/batch/XMLParserTest.java]: rename it to transforproperties

File: cdap-integration-tests/integration-test-remote/src/test/java/co/cask/cdap/app/etl/batch/XMLParserTest.java [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+import java.util.concurrent.TimeUnit;
+
+public class XMLParserTest extends ETLTestBase {
+  public static final String XMLPARSER_SOURCE = "xmlParserSource";
+  public static final String XMLPARSER_SINK = "xmlParserSink";
+
+  private static final Schema SOURCE_SCHEMA =
+    Schema.recordOf("inputRecord",
+                    Schema.Field.of("offset", Schema.of(Schema.Type.INT)),
+                    Schema.Field.of("body", Schema.of(Schema.Type.STRING)));
```
Comment (bhushankawadkar) [inline cdap-integration-tests/integration-test-remote/src/test/java/co/cask/cdap/app/etl/batch/XMLParserTest.java]: Use XMLReader schema as offset, filename and record.

File: cdap-integration-tests/integration-test-remote/src/test/java/co/cask/cdap/app/etl/batch/XMLParserTest.java [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+  private void ingestInputData(String inputDatasetName) throws Exception {
+    DataSetManager<Table> inputManager = getTableDataset(inputDatasetName);
+    Table inputTable = inputManager.get();
+
+    putValues(inputTable, 1, 1, "<bookstore><book category=\"cooking\"><subcategory><type>Continental</type>" +
+      "</subcategory><title lang=\"en\">Everyday Italian</title><author>Giada De Laurentiis</author>" +
+      "<year>2005</year><price>30.00</price></book></bookstore>");
+    putValues(inputTable, 2, 2, "<bookstore><book category=\"children\"><subcategory><type>Series</type>" +
+      "</subcategory><title lang=\"en\">Harry Potter</title><author>J K. Rowling</author>" +
+      "<year>2005</year><price>49.99</price></book></bookstore>");
```
Comment (bhushankawadkar) [inline cdap-integration-tests/integration-test-remote/src/test/java/co/cask/cdap/app/etl/batch/XMLParserTest.java]: put one error record and check it in assert

**Comment by bhushankawadkar** [comment]
LGTM

**Comment by ameyk** [comment]
merge(if not already) and close PR if reviewed

**Pull request merged**